src/http_interface.py

```diff
@@ -174,6 +174,28 @@ def get_services_definition(self) -> dict:
         return legacy.get_services_from_relation_data(relation_data)
 
 
+class HTTPProvider(_IntegrationInterfaceBaseClass):
+    """HTTP interface provider class to be instantiated by the haproxy-operator charm."""
+
+    def _on_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Handle relation-changed event.
+
+        Args:
+            event: relation-changed event.
+        """
+        event.relation.data[self.charm.unit].update({"hostname": self.bind_address, "port": 80})
+
+    # We add a placeholder implementation of this method because of parent class
+    def _on_relation_changed(self, _: RelationChangedEvent) -> None:
+        """Handle relation-changed event."""
+        logger.warning("Nothing to do for relation-changed hook of website relation, skipping.")
```

Comment:
Not sure a warning is required here, that's nothing abnormal, "info" or "debug" are better suited.
Same for relation broken

Reply:
updated to "debug"